preprocessing/shift_generation.py

```diff
@@ -184,29 +205,30 @@ def get_shifts_overlapping_t(shifts, time_sets):
     return shifts_overlapping_t
 
 
-def get_off_shifts(root):
+def get_off_shifts(shifts_per_week):
 
-    events = get_start_events(root)
-
-    off_shifts = []
+    off_shifts = tuplelist()
     off_shifts_in_week = tupledict()
-    week = 0
-    off_shifts_in_week[week] = []
-
-    for i in range(len(events)):
-        for event in events[i:]:
-            duration = event - events[i]
-            if events[i] >= (week + 1) * 24 * 7:
-                week += 1
-                off_shifts_in_week[week] = []
-            if event >= (week + 1) * 24 * 7:
-                break
-            if duration > 70:
-                break
-            elif duration >= 36:
-                if (events[i], duration) not in off_shifts:
-                    off_shifts_in_week[week].append((events[i], duration))
-                    off_shifts.append((events[i], duration))
+
+    for week in shifts_per_week:
```

Comment:
It is better to only extract the keys with `.keys() when you do not need the values, or to extract both key and value with 
```python
for key, value in dict_name.items()
```

Seems to me that the latter might be better, as it can simplify a lot of lines.

Reply:
Not familiar with this way of for-looping. Could test it!

Reply:
This is done!